### ISSUE TITLE
[BUG] DeviceSearchQuery export can't get job progress?

### ISSUE THREAD

**Issue by ecgg** [carbonblack/carbon-black-cloud-sdk-python#486]
**I am seeing this behaviour on: (please complete the following information):**
 - OS: Red Hat Enterprise Linux Server release 7.9 (Maipo)
 - Carbon Black Cloud Products: cbc_sdk 1.5.1
 - Python Version: 3.10.11

**Describe the bug**

With DeviceSearchQuery.download() now being deprecated in favour of export, I adapted code to make use of it using [Live Query Export Results](https://carbon-black-cloud-python-sdk.readthedocs.io/en/latest/live-query/#export-results) as an example. 

When await_completion() checks for progress, it seems to hit a non-existing resource (HTTP 404). When get_result() is called on the Future Job, it results in a traceback

**Steps to Reproduce**
 ```
   try:
        query = cb_api.select(Device)
        query = query.set_status(["ALL"])
        # devs = query.download()
        job = query.export()
        logging.debug("job %s - %s", job.id, job.status)
        job_future = job.await_completion()
        finished_job = job_future.result()
        finished_job.refresh()
        devs = finished_job.get_output_as_string()
    except ClientError as e:
        logging.error("Client error: %s", e)
```

**Expected behavior**
Get progress of running async job

**Screenshots**
Logging output

```
DEBUG Sending HTTP POST /appservices/v6/orgs/ORGKEY/devices/_export with {"criteria": {"status": ["ALL"]}, "format": "CSV"}
DEBUG Starting new HTTPS connection (1): URL
DEBUG https://URL "POST /appservices/v6/orgs/ORGKEY/devices/_export HTTP/1.1" 303 38
DEBUG https://URL "GET /jobs/v1/orgs/ORGKEY/jobs/2358531 HTTP/1.1" 200 200
DEBUG HTTP POST /appservices/v6/orgs/ORGKEY/devices/_export took 0.039s (response 200)
DEBUG job 2358531 - IN_PROGRESS
DEBUG https://URL "GET /jobs/v1/orgs/ORGKEY/jobs/2358531/progress HTTP/1.1" 404 117
DEBUG HTTP GET /jobs/v1/orgs/ORGKEY/jobs/2358531/progress took 0.038s (response 404)
DEBUG https://URL "GET /jobs/v1/orgs/ORGKEY/jobs/2358531/progress HTTP/1.1" 404 117
DEBUG HTTP GET /jobs/v1/orgs/ORGKEY/jobs/2358531/progress took 0.039s (response 404)
DEBUG https://URL "GET /jobs/v1/orgs/ORGKEY/jobs/2358531/progress HTTP/1.1" 404 117
DEBUG HTTP GET /jobs/v1/orgs/ORGKEY/jobs/2358531/progress took 0.041s (response 404)
Traceback (most recent call last):
  File "cb_151.py", line 66, in <module>
    finished_job = job_future.result()
  File "3.10.11/lib/python3.10/concurrent/futures/_base.py", line 458, in result
    return self.__get_result()
  File "3.10.11/lib/python3.10/concurrent/futures/_base.py", line 403, in __get_result
    raise self._exception
  File "3.10.11/lib/python3.10/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "v3.10.11_carbonblack/lib/python3.10/site-packages/cbc_sdk/platform/jobs.py", line 128, in <lambda>
    return self._cb._async_submit(lambda arg, kwarg: arg[0]._await_completion(), self)
  File "v3.10.11_carbonblack/lib/python3.10/site-packages/cbc_sdk/platform/jobs.py", line 105, in _await_completion
    progress_data = self.get_progress()
  File "v3.10.11_carbonblack/lib/python3.10/site-packages/cbc_sdk/platform/jobs.py", line 84, in get_progress
    resp = self._cb.get_object(url)
  File "v3.10.11_carbonblack/lib/python3.10/site-packages/cbc_sdk/connection.py", line 479, in get_object
    result = self.api_json_request("GET", uri, params=query_parameters)
  File "v3.10.11_carbonblack/lib/python3.10/site-packages/cbc_sdk/connection.py", line 554, in api_json_request
    result = self.session.http_request(method, uri, headers=headers, data=raw_data, **kwargs)
  File "v3.10.11_carbonblack/lib/python3.10/site-packages/cbc_sdk/connection.py", line 326, in http_request
    raise ObjectNotFoundError(uri=uri, message=r.text)
cbc_sdk.errors.ObjectNotFoundError: Received 404 (Object Not Found) for https://URLt/jobs/v1/orgs/ORGKEY/jobs/2358531/progress: {"error_code":"PROGRESS_NOT_FOUND","message":"No progress available for job 2358531","job_id":2358531}
```

**Additional context**

In the GUI, we see the export requests showing up under notifications, however it mentions 'Couldn't download - no file'

![gui](https://github.com/carbonblack/carbon-black-cloud-sdk-python/assets/27764963/49765b9c-9cb7-4328-93c5-6abba3509fd3)

**Comment by abowersox-cb** [comment]
I've looked into this using Postman to trigger the API requests manually.  From what I can tell, the "Job Progress" API call is not working correctly for jobs created by a "Device Export" API call. The "Job Details" and "Job Download" API calls, however, work fine.  This may be the cause of the weirdness you see in the UI as well; if the UI relies on the Job Progress call, it might not be able to recognize when a job is completed.

I have reported this as a bug in the server.  There _may_ be a way to work around this at the SDK level; if you're interested, I could spend a little time to hash out something.

**Comment by abowersox-cb** [comment]
I have come up with a workaround that provides a way to wait on a Job that references the job details status instead of the job progress, thus avoiding the buggy API call.  Here's a complete script you can use.

```
#!/usr/bin/env python

import sys
import time
import logging
from cbc_sdk.helpers import build_cli_parser, get_cb_cloud_object
from cbc_sdk.platform import Device
from cbc_sdk.errors import ApiError, ClientError


def _await_completion_on_details(job):
	"""
	Waits for this job to complete by examining the details status.

	Required Permissions:
		jobs.status(READ)
		
	Args:
		job (Job): The job to wait on.

	Returns:
		Job: The job that was passed in.
	"""
	status = ""
	do_sleep = False
	errorcount = 0
	while status not in ("FAILED", "COMPLETED"):
		if do_sleep:
			time.sleep(0.5)
		try:
			job._refresh()
			if job.status != status:
				status = job.status
		except (ServerError, ObjectNotFoundError):
			errorcount += 1
			if errorcount == 3:
				raise
			status = ""
		do_sleep = True
	if status == "FAILED":
		raise ApiError(f"Job {job.id} reports failure")
	return job

		
def await_completion_on_details(job):
	"""
	Create a Python Future to check for job completion and return results when available.

	Returns a Future object which can be used to await results that are ready to fetch. This function call
	does not block.

	Required Permissions:
		jobs.status(READ)

	Args:
		job (Job): The job to wait on.

	Returns:
		Future: A future which can be used to wait for this job's completion. When complete, the result of the
				Future will be this object.
	"""
	return job._cb._async_submit(lambda arg, kwarg: _await_completion_on_details(arg[0]), job)


def main():
	"""Main function to demo the workaround"""
	parser = build_cli_parser("New way of awaiting completion of device export job")
	args = parser.parse_args()
	cb_api = get_cb_cloud_object(args)
	
	# code sequence from ecgg, revised
	try:
		query = cb_api.select(Device)
		query = query.set_status(["ALL"])
		# devs = query.download()
		job = query.export()
		logging.debug("job %s - %s", job.id, job.status)
		job_future = await_completion_on_details(job)
		finished_job = job_future.result()
		finished_job.refresh()
		devs = finished_job.get_output_as_string()
		print(devs)
	except ClientError as e:
		logging.error("Client error: %s", e)
	return 0


if __name__ == '__main__':
	sys.exit(main())
```
We will likely be looking to add this into future versions of the SDK to work around this problem, but this should help you out now.

**Comment by ecgg** [comment]
Workaround works perfectly - thank you!

**Comment by abowersox-cb** [comment]
Issue closed for now; more-permanent fix has been added to internal JIRA for 1.5.2 release.
